Src/Plugins/ValidatorBase.cs

```diff
@@ -199,6 +212,15 @@ public static string ParseExpression(IRegex regexEngine, string matchedPattern,
             return match?.Success ?? false ? ParseValue(match.Value) : null;
         }
 
+        public virtual void MatchCleanup(ref string matchedPattern,
```

Comment:
>MatchCleanup [](start = 28, length = 12)

I don't see the utility of dividing the static analysis into two phases like this. it isn't the change that i proposed. I'd prefer not to do this. The reason is that this division between an initial clean-up and the rest of the code is non-obvious and not necessary for the bulk of validators. this OM change doesn't belong in the base class, therefore.

Reply:
Removed.

---
In reply to: [575504321](https://github.com/microsoft/sarif-pattern-matcher/pull/189#discussion_r575504321) [](ancestors = 575504321)